app/services/notifications/create_from_model_service/work_package_strategy.rb

```diff
@@ -31,8 +31,14 @@ def self.reasons
     %i(mentioned assigned responsible watched commented created processed prioritized scheduled shared)
   end
 
-  def self.permission
-    :view_work_packages
+  def self.permission(journal, _reason)
+    if journal.restricted?
```

Comment:
Just noted journal can be nil, in other contexts. Safe navigate
```suggestion
    if journal&.restricted?
```